### PR TITLE
Connect wallet button [interchain UI part 2]

### PR DESCRIPTION
**Pull request by grod220**
Secures connection approval from user intending to do an IBC in. Also, the recent updates of interchain ui seem to have broken the dropdown component. Went ahead and swapped it out for shadcn/ui components.

https://github.com/penumbra-zone/web/assets/16624263/a837af48-73ff-4d32-8e1f-21c65392d1b1

### CODE REVIEW

File: apps/minifront/src/components/ibc/ibc-in/chain-dropdown.tsx
Comment (grod220) [inline apps/minifront/src/components/ibc/ibc-in/chain-dropdown.tsx:1]: Component completely broken after interchain deps update. Went ahead and swapped it out for https://ui.shadcn.com/docs/components/combobox

File: apps/minifront/src/components/ibc/ibc-in/chain-provider.tsx [hunk 1]
```diff
@@ -55,6 +55,7 @@ export const IbcChainProvider = ({ registry, children }: IbcProviderProps) => {
         },
       }}
       signerOptions={signerOptions}
+      modalTheme={{ defaultTheme: 'light' }}
```
Comment (grod220) [inline apps/minifront/src/components/ibc/ibc-in/chain-provider.tsx:58]: Looks like the default changed to dark. Forcing the light theme here.

File: apps/minifront/src/components/ibc/ibc-in/wallet-connect-button.tsx
Comment (grod220) [inline apps/minifront/src/components/ibc/ibc-in/wallet-connect-button.tsx:1]: Starting point: https://github.com/cosmology-tech/create-cosmos-app/blob/main/examples/asset-list/components/wallet/Wallet.tsx
Lifted, simplified, and changed to tailwind

File: apps/minifront/src/components/ibc/layout.tsx [hunk 2]
```diff
@@ -8,12 +8,12 @@ export const IbcLayout = () => {
     <>
       <div className="fixed inset-0 z-[-100] size-full bg-[url('penumbra-logo.svg')] bg-[length:160vmax] bg-fixed bg-[top_50%_left_25vw] bg-no-repeat" />
       <div className='flex flex-1 flex-col gap-4 md:flex-row md:place-content-around'>
-        <Card light className='relative overflow-visible md:self-start'>
+        <Card light className='relative z-10 overflow-visible md:self-start'>
```
Comment (grod220) [inline apps/minifront/src/components/ibc/layout.tsx:11]: So arrows don't overlap on top

File: apps/minifront/src/components/ibc/layout.tsx [hunk 3]
```diff
@@ -8,12 +8,12 @@ export const IbcLayout = () => {
...
           <LongArrowIcon
             direction='right'
             // Negative calculated margin giving lint issue
             /* eslint-disable-next-line tailwindcss/enforces-negative-arbitrary-values */
-            className='invisible absolute inset-y-0 right-0 my-auto -mr-[calc(30vw-3px)] size-[30vw] text-stone-300 md:visible'
+            className='invisible absolute -top-44 right-0 z-0 -mr-[calc(30vw-3px)] size-[30vw] text-stone-300 md:visible'
```
Comment (grod220) [inline apps/minifront/src/components/ibc/layout.tsx:16]: Relocated the arrows to be fixed versus floating in the middle of the card. That way, as the card changes size the arrows don't move.

**Review by Valentine1898 — APPROVED** [review]
LGTM! 
I think that we should strive to make the IBC interface more intuitive for the user (like it is done by osmosis) 
https://jam.dev/c/927989bd-68f2-4762-b8ef-c656738a2b02

**Comment by grod220** [comment]
Definitely agree @Valentine1898, ideally we can later have a designer make this flow much nicer

**Pull request merged**